x-pack/plugins/siem/public/timelines/components/timeline/body/events/event_column_view.tsx

```diff
@@ -79,17 +86,70 @@ export const EventColumnView = React.memo<Props>(
     toggleShowNotes,
     updateNote,
   }) => {
-    const timelineTypeContext = useTimelineTypeContext();
-
+    const { getManageTimelineById } = useManageTimeline();
+    const timelineActions = useMemo(() => getManageTimelineById(timelineId).timelineRowActions, [
+      getManageTimelineById,
+      timelineId,
+    ]);
     const additionalActions = useMemo<JSX.Element[]>(() => {
-      return (
-        timelineTypeContext.timelineActions?.map((action) => (
-          <EventsTdContent key={action.id} textAlign="center">
-            {action.getAction({ eventId: id, ecsData })}
-          </EventsTdContent>
-        )) ?? []
+      const grouped = timelineActions.reduce(
+        (
+          acc: {
+            contextMenu: JSX.Element[];
+            icon: JSX.Element[];
+          },
+          action
+        ) => {
+          if (action.displayType === 'icon') {
+            return {
+              ...acc,
+              icon: [
+                ...acc.icon,
+                <EventsTdContent key={action.id} textAlign="center">
+                  <EuiToolTip
+                    data-test-subj={`${action.dataTestSubj}-tool-tip`}
+                    content={action.content}
+                  >
+                    <EuiButtonIcon
+                      aria-label={action.ariaLabel}
+                      data-test-subj={`${action.dataTestSubj}-button`}
+                      iconType={action.iconType}
+                      isDisabled={action.isActionDisabled ?? false}
+                      onClick={() => action.onClick({ eventId: id, ecsData })}
+                    />
+                  </EuiToolTip>
+                </EventsTdContent>,
+              ],
+            };
+          }
+          return {
+            ...acc,
+            contextMenu: [
+              ...acc.contextMenu,
+              <EuiContextMenuItem
```

Comment:
I think there's specific styling for this popover since the timeline text is a bit smaller at `12px`. Here's the current [mocks](https://www.figma.com/file/jcCKnGXvOlFxMOpUjlTMMz/All-Exceptions?node-id=490%3A13048), which have the text size at `14px` instead of the default `16px`. Though the Timeline text in that mock is at `14px`, so I think we'll need to verify with design -- @marrasherrier, can you comment on if we should override the default styles for this `EuiPopover`?